library/core/src/num/bignum.rs

```diff
@@ -162,20 +162,13 @@ macro_rules! define_bignum {
                 let digits = self.digits();
                 let zeros = digits.iter().rev().take_while(|&&x| x == 0).count();
                 let end = digits.len() - zeros;
```

Comment:
(I'm new to this code, so let me know if I'm misunderstanding things.  It took me a bit to figure out what was going on.)

I think this could be simplified further?

If I'm reading it right, I think this is something like
```rust
let digits = self.digits();
// Find the most significant non-zero digit
let msd = digits.iter().rposition(|&x| x != 0);
if let Some(msd) = msd {
    let digitbits = <$ty>::BITS as usize;
    msd * digitbits + digits[msd].log2() as usize
} else {
    // There are no non-zero digits, i.e., the number is zero.
    return 0;
}
```

Reply:
I think the `log2()` would be slower (more complex instruction that does a lot we don't need) than a CLZ, but the rest of the changes I can certainly incorporate. Thanks!

Reply:
`log2` on integers is just `BITS - CLZ`, though: https://github.com/rust-lang/rust/blob/23ce5fc4655ed546f74a85fc8836e95bec0c64fd/library/core/src/num/uint_macros.rs#L841-L849

(Oh, I might be off-by-one in the code.  Up to you if you like `... + log2() + 1` or `(msd + 1) * digitbits - leading_zeros()` or whatever best.)

Reply:
Oh, my bad, I thought `log2()` would return the actual logarithm base 2, not just the integer part. I'll totally use that :)

Reply:
> `log2` on integers is just `BITS - CLZ`, though:
> 
> https://github.com/rust-lang/rust/blob/23ce5fc4655ed546f74a85fc8836e95bec0c64fd/library/core/src/num/uint_macros.rs#L841-L849
> 
> (Oh, I might be off-by-one in the code. Up to you if you like `... + log2() + 1` or `(msd + 1) * digitbits - leading_zeros()` or whatever best.)

Yep! We're always off-by-one the first time or two. :) I worked it out.

I went with a `match` statement instead of `if let` since it is one line shorter.